src/snowflake/cli/api/project/schemas/entities/application_package_entity.py

```diff
@@ -0,0 +1,50 @@
+# Copyright (c) 2024 Snowflake Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import ClassVar, List, Optional, Union
+
+from pydantic import Field
+from snowflake.cli.api.project.schemas.entities.common import (
+    EntityBase,
+    EntityType,
+)
+from snowflake.cli.api.project.schemas.native_app.package import DistributionOptions
+from snowflake.cli.api.project.schemas.native_app.path_mapping import PathMapping
+
+
+class ApplicationPackageEntity(EntityBase):
```

Comment:
There are some properties not accounted for here, like `scratch_stage`, `generated_root`, and `bundle_root`. What's the plan for those?

Reply:
Added these as well. While we maintain both versions we will have to remember to update v2 schema when we make v1 changes.